samples/cellular/lwm2m_client/fota_external_mcu.rst

```diff
@@ -0,0 +1,285 @@
+
+nRF52840 configuration
+======================
+
+:ref:`smp_svr` have reference implementation that uses MCUBoot recovery mode or serial SMP server mode to allow image update.
+
+Build
+-----
+
+To build with MCUBoot recovery mode
+
+.. code-block:: console
+
+   west build --pristine -b nrf9160dk_nrf52840 -- -DEXTRA_DTC_OVERLAY_FILE:="nrf9160dk_nrf52840_recovery.overlay"
+
+To build with Native serial MCUMgr server for UART1
+
+.. code-block:: console
+
+   west build --pristine -b nrf9160dk_nrf52840 -- -DOVERLAY_CONFIG="overlay-serial.conf" -DEXTRA_DTC_OVERLAY_FILE:="nrf9160dk_nrf52840_mcumgr_srv.overlay"
+
+Verify image update
+===================
+
+Following flow uses generic MCUmgr client mode and Coiote without bootstrap.
+
+#. To set up the script, you must set the username and password that you used in AVSystem's Coiote Device Management server as the environment variables.
+
+   .. code-block:: console
+
+      # Setup phase
+      export COIOTE_PASSWD='my-password'
+      export COIOTE_USER='my-username'
+
+#. Build and flash nRF52840 smp_svr sample:
+
+   Please flip the DK board's PROG/DEBUG switch to nRF52 position.
+
+   .. code-block:: console
+
+      cd samples/cellular/smp_svr/
+      west build --pristine -b nrf9160dk_nrf52840 -- -DOVERLAY_CONFIG="overlay-serial.conf" -DEXTRA_DTC_OVERLAY_FILE:="nrf9160dk_nrf52840_mcumgr_srv.overlay"
+      west flash --erase
+
+#. Open the :file:`src/prj.conf` file.
+
+#. Change :kconfig:option:`CONFIG_MCUBOOT_IMAGE_VERSION` to ``1.1.0``.
+
+#. Rebuild the sample and copy binary to lwm2m_client sample folder:
+
+   .. code-block:: console
+
+      west build
+      cp build/zephyr/app_update.bin ../lwm2m_client/app_update52.bin
+      cd ..
+
+#. Open terminal connection for nRF52840 (VCOM1) at linux /dev/ttyACM1 and call `mcuboot` command on the shell.
+
+   .. code-block:: console
+
+      uart:~$ mcuboot
+      swap type: none
+      confirmed: 1
+
+      primary area (2):
+         version: 1.0.0+0
+         image size: 70136
+         magic: unset
+         swap type: none
+         copy done: unset
+         image ok: unset
+
+      failed to read secondary area (5) header: -5
+      uart:~$
+
+#. Register your nRF9160 device with the Coiote Device management server.
+
+#. Build and flash lwm2m_client sample.
+
+   Please flip the DK board's PROG/DEBUG switch to nRF91 position.
+
+   .. code-block:: console
+
+      cd lwm2m_client/
+      west build  --pristine -b nrf9160dk_nrf9160_ns --  -DOVERLAY_CONFIG="overlay-adv-firmware.conf;overlay-fota_helper.conf;overlay-avsystem.conf;overlay-lwm2m-1.1.conf;overlay-mcumgr_client.conf" -DEXTRA_DTC_OVERLAY_FILE="nrf9160dk_mcumgr_client_uart2.overlay"
+      west flash --erase
+
+#. Wait for the device registration to be complete.
+
+#. nRF52840 firmware by using the :file:`/scripts/fota.py` script.
```

Comment:
Something seems to be missing here.
Maybe something like:

"Use the ... script to update the nRF52840 SoC firmware."